src/config_files/build.rs

```diff
@@ -0,0 +1,58 @@
+use base64;
+use colored::*;
+use error::DefaultResult;
+use serde_json;
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::Read,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+#[derive(Clone, Deserialize)]
+pub struct Build {
+    pub steps: HashMap<String, Vec<String>>,
+    pub artifact: PathBuf,
+}
+
+impl Build {
+    pub fn from_file<T: AsRef<Path>>(path: T) -> DefaultResult<Build> {
+        let file = File::open(path)?;
+
+        let build = serde_json::from_reader(&file)?;
+
+        Ok(build)
+    }
+
+    /// Starts the build using the supplied build steps and returns the contents of the artifact
```

Comment:
This is very nice solution for building @sphinxc0re! Just one issue.... :-)  Tests!!   I know that this is tricky in the context of code that fires up commands, and in the go version of hcdev I started without writing lots of tests for that, which came back to bite my hindquarters.   

One important integration test, I think, is that the `init` command which produces a scaffold app, should also produce valid .build files, so we should be able to have a test that both builds that directory, packages it all up and fires up a holochain instance with it.  End-to-end.

Reply:
For this ticket, I think lets just please add one test that at least confirms that build.run() actually does execute the command in the steps.

Reply:
Yeah, what @zippy said. Great work!
I've added .build files to app-spec-rust here: https://github.com/holochain/app-spec-rust/pull/1/commits/add8ce23cbe997fd730363dcf8badd66bc03024f, which is kinda an integration test. (which works! :))

But yeah, would be good to have a unit test in here as well, I agree. I nevertheless approve this PR since Eric's change request is blocking the merge already. Feel free to merge after adding a test that satisfies him ;)

Reply:
The problem here is that having a `.build` file in a code directory makes the bundle non-reversible. One of the tests ensures, that `package` and `unpack` are symmetric. I would like to finish up the generate command and make that test the build steps.